std/base64.d

```diff
@@ -390,9 +390,8 @@ template Base64Impl(char Map62th, char Map63th, char Padding = '=')
      * Returns:
      *  The number of times the output range's $(D put) method was invoked.
      */
-    size_t encode(R1, R2)(in R1 source, auto ref R2 range)
-        if (isArray!R1 && is(ElementType!R1 : ubyte) &&
-            !is(R2 == char[]) && isOutputRange!(R2, char))
+    size_t encode(E, R)(scope const(E)[] source, auto ref R range)
+    if (is(E : ubyte) && isOutputRange!(R, char))
```

Comment:
char[] is now allowed. What's the motivation for this?

Reply:
`char[]` is now allowed to be the output range. I see no reason to disallow it if `isOutputRange!(R, char)` is `true`

Reply:
Ok, but maybe we should add a test for `char[]` then?

Reply:
`isOutputRange!(char[], char) == false` due to auto-decoding.